samples/ReverseProxy.Auth.Sample/TokenService.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Yarp.Sample
+{
+    internal class TokenService
+    {
+        internal Task<string> GetAuthTokenAsync(ClaimsPrincipal user)
+        {
+            // we only have tokens for bob
+            if (string.Equals("Bob", user.Identity.Name))
+            {
+                return Task.FromResult("valid");
+            }
+            return Task.FromResult<string>(null);
+        }
+    }
```

Comment:
Although a sample, but it should show a perf-concise way with
```suggestion
    internal class TokenService
    {
        private static readonly Task<string> s_validTask = Task.FromResult("valid");
        
        internal Task<string> GetAuthTokenAsync(ClaimsPrincipal user)
        {
            // we only have tokens for bob
            if (string.Equals("Bob", user.Identity.Name))
            {
                return s_validTask;
            }
            return Task.FromResult<string>(null);
        }
    }
```
to save the repeated task allocation (72B on 64-bit)?

It's more an educational thing for people that aren't aware of such things.

Reply:
It shouldn't actually be a static value for security reasons. Maybe I should change it to new Guid to clarify that.

Reply:
👍🏻 even better!